inconsistencies.md

```diff
@@ -3,140 +3,261 @@
 This plan makes the following assumption:
 
   - We are not willing to rewrite mailbox/cassandra with event sourcing
-  - We will rather identify projections and how to heal them
+  - We will try to identify possibly inconsistent duplicated information, and how to resolve it.
 
-# Action plan
-
-(prioritized) this section present the conclusion of this studu which details can be found below.
-
-## Grooming
-
-Performance impact: lightweight transaction VS counters
-
-## High
-
- - SolveInconsistencies webadmin tasks:
-   - messages (MessageIdToImapUid -> MessageIdTable)
-   - ACLs (ACL/CassandraUserMailboxRightsTable)
-   - Mailbox counters (/!\ counters)
-   - Current quota (/!\ counters)
+# Details
 
- - Testing:
-   - MessageIdToImapUid -> MessageIdTable
-   - Message <-> attachment
-   - Upload attachment
+## mailbox/mailboxPath
 
- - Retry:
-   - messages (MessageIdToImapUid -> MessageIdTable)
+Table: mailbox
+Table: mailboxPathV2
 
-### Low
+Why: This projection is used for reading mailboxes by mailboxPath or by mailboxId. We also ensure mailboxPath unicity.
 
-Used to speed up IMAP operation (SELECT / EXPUNGE)
+Operation:
+ - Create: We write to `mailboxPathV2` in order to ensure path unicity, then write to `mailbox`. If partially applied,
+ then we can have an orphan `mailboxPathV2`. The corresponding mailbox don't exist, but can't either be created.
+ - Rename: We write the new path to `mailboxPathV2` in order to ensure path unicity, then we read the old path from 
+ `mailbox` (op1) to remove it from `mailboxPathV2` (op2) and finally we update the `mailbox` to reference it with 
+ new path (op3). If op1, op2 or op3 fails, we have an entry conflict that might require merging the mailboxes.
+ - Delete: We remove the `mailbox` entry, and finally the `mailboxPathV2`. If partially applied,
+ then we can have an orphan `mailboxPathV2`. The corresponding mailbox don't exist, but can't either be created.
```

Comment:
don't we have a design issue here? mailboxPathV2 should be a projection but keep the truth about the existence of the mailbox. While mailbox is the main table which can be used to rebuild the projection(s) but is not seen as the table of truth.

Reply:
That's a specificity here yes. Sounds like the source of truth is `mailbox` as it contains the UID_VALIDITY. Also an immutable identifier sounds like also a desirable property from a source of truth.

Nevertheless we have an invariant on the name that we need to apply. That's why all of this looks a bit brittle.

But the good news is that's it is a behavior that only happen for mailboxes.

Reply:
now that we see there are issues related, see https://issues.apache.org/jira/browse/JAMES-3074 what about having UID_VALIDITY in the mailboxPath table instead?

Reply:
Also an immutable identifier sounds like also a desirable property from a source of truth.

Reply:
Naming here is the least of my concerns ;-)